helix-vcs/src/git.rs

```diff
@@ -41,16 +48,8 @@ impl Git {
             ..gix::Permissions::default_for_level(gix::sec::Trust::Full)
         });
 
-        let mut open_options = gix::discover::upwards::Options::default();
-        if let Some(ceiling_dir) = ceiling_dir {
-            open_options.ceiling_dirs = vec![ceiling_dir.to_owned()];
```

Comment:
I actually recommend to not go ahead with this PR and instead use the new `dot_git_only` option. With it set to `true` the discovery can be optimal for this usecase, while making it easy to setup ceiling dirs and handle cross-filesystem checks.

Reply:
thanks for  taking this further, I will update this PR to use the new gitoxide version once that is released